src/main/java/io/github/genomicdatainfrastructure/daam/gateways/RemsApplicationMapper.java

```diff
@@ -321,4 +324,15 @@ private List<Label> toLabelObject(Map<String, String> map) {
                 .map(entry -> new Label(entry.getKey(), entry.getValue()))
                 .toList();
     }
+
+    private List<Label> toLabelObjectFromLong(Map<String, Long> map) {
```

Comment:
important: to avoid dupplicated code, you should generify `toLabelObject()`, instead of adding `toLabelObjectFromLong()`. 

If you change the signature of the method to `toLabelObject(Map<String, Object> map)`, and call `entry.getValue().toString()` would be enough.

I would also check if `entry.getValue()` is not nullable, before calling `.toString()`. Keys can never be null, but values can.

Reply:
I did it toLabelObject(Map<String, ?> map) so it can be more flexible, or is it too broad?

Reply:
Nope, it's excelent.